presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -252,6 +242,10 @@ public WindowOperator(
             this.ordering = ImmutableList.copyOf(concat(nCopies(unGroupedPartitionChannels.size(), ASC_NULLS_LAST), sortOrder));
         }
 
+        this.outputPages = WorkProcessor.create(new ProducePagesIndexes(operatorContext.aggregateUserMemoryContext()))
+                .flatMap(new ProduceWindowPartitions())
```

Comment:
Convert `ProduceWindowPartitions` into a method (this will make that code less nested) and call it like this here:
```
.flatMap(this::produceWindowPartitions)
```

or even:
```
// here ProduceWindowPartitions is is just the inner (currently anonymous) class as returned by (current) ProduceWindowPartitions#apply
.flatMap(pagesIndex -> new ProduceWindowPartitions(pagesIndex)
```

Reply:
Hm. My thinking is that either all elements are methods or classes, so it's consistent

Reply:
i understand. OTOH, this reduces code nesting significantly (because this one is not a Process, it's a flat-map's Function which returns sub-WorkProcessors)